sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataAggregator.java

```diff
@@ -125,6 +143,17 @@ static class EventDataAggregatorMain implements Subscription, CoreSubscriber<Eve
                 });
         }
 
+        /**
+         * The number of events in the current batch.
+         *
+         * @return Number of events in the current batch.
+         */
+        public int getNumberOfEventsInCurrentBatch() {
+            final EventDataBatch b = currentBatch;
```

Comment:
given `getNumberOfEventsInCurrentBatch()` is potentially not protected by drain-loop, should we lock the access so it sync with the access in `updateOrPublishBatch`?

```java
final EventDataBatch b;
synchronized (lock) {
   b = currentBatch;
}
return b != null ? b.getCount() : 0;
```